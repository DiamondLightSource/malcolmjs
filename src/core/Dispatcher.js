/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
