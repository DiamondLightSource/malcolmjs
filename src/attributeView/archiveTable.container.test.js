import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import MockCircularBuffer from '../malcolm/reducer/attribute.reducer.mocks';
import ArchiveTable from './archiveTable.container';

describe('archiveTable', () => {
  let shallow;
  let mockArchive;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    mockArchive = {
      timeStamp: new MockCircularBuffer(5),
      value: new MockCircularBuffer(5),
      alarmState: new MockCircularBuffer(5),
    };
  });

  it('renders correctly', () => {
    mockArchive.timeStamp.push(new Date(1000));
    mockArchive.timeStamp.push(new Date(2000));
    mockArchive.timeStamp.push(new Date(3000));
    mockArchive.timeStamp.push(new Date(3000));
    mockArchive.value.push(1);
    mockArchive.value.push(-1);
    mockArchive.value.push(false);
    mockArchive.value.push(false);
    mockArchive.alarmState.push(0);
    mockArchive.alarmState.push(1);
    mockArchive.alarmState.push(0);
    mockArchive.alarmState.push(0);

    const wrapper = shallow(<ArchiveTable attribute={mockArchive} />);
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
