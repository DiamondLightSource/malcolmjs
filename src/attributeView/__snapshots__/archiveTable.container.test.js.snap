// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`archiveTable renders correctly 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <WithStyles(Table)
    className="WidgetTable-headerLayoutNoScroll-3"
  >
    <WithStyles(TableHead)>
      <WithStyles(TableRow)
        className="WidgetTable-rowFormat-6"
      >
        <WithStyles(TableCell)
          className="WidgetTable-textHeadings-8"
          key="0"
          padding="none"
        >
          Alarm state
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          className="WidgetTable-textHeadings-8"
          key="1"
          padding="none"
        >
          Time set
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          className="WidgetTable-textHeadings-8"
          key="2"
          padding="none"
        >
          Value
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
  </WithStyles(Table)>
  <div
    className="WidgetTable-tableBody-5"
  >
    <WithStyles(Table)
      className="WidgetTable-tableLayout-4"
    >
      <WithStyles(TableBody)>
        <WithStyles(TableRow)
          className="WidgetTable-rowFormat-6"
          key="0"
          onClick={[Function]}
        >
          <WithStyles(TableCell)
            className="WidgetTable-textBody-10"
            key="0,0"
            padding="none"
          >
            <WithTheme(TableWidgetSelector)
              columnMeta={
                Object {
                  "label": "Alarm state",
                  "tags": Array [
                    "info:alarm",
                  ],
                }
              }
              columnWidgetTag="info:alarm"
              rowChangeHandler={[Function]}
              rowPath={
                Object {
                  "column": 0,
                  "label": "alarm",
                  "row": 0,
                }
              }
              setFlag={[Function]}
              value={0}
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            className="WidgetTable-textBody-10"
            key="0,1"
            padding="none"
          >
            <WithTheme(TableWidgetSelector)
              columnMeta={
                Object {
                  "label": "Time set",
                  "tags": Array [
                    "widget:textupdate",
                  ],
                }
              }
              columnWidgetTag="widget:textupdate"
              rowChangeHandler={[Function]}
              rowPath={
                Object {
                  "column": 1,
                  "label": "timeStamp",
                  "row": 0,
                }
              }
              setFlag={[Function]}
              value="1970-01-01T00:00:01.000Z"
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            className="WidgetTable-textBody-10"
            key="0,2"
            padding="none"
          >
            <WithTheme(TableWidgetSelector)
              columnMeta={
                Object {
                  "label": "Value",
                  "tags": Array [
                    "widget:textupdate",
                  ],
                }
              }
              columnWidgetTag="widget:textupdate"
              rowChangeHandler={[Function]}
              rowPath={
                Object {
                  "column": 2,
                  "label": "value",
                  "row": 0,
                }
              }
              setFlag={[Function]}
              value={1}
            />
          </WithStyles(TableCell)>
        </WithStyles(TableRow)>
        <WithStyles(TableRow)
          className="WidgetTable-rowFormat-6"
          key="1"
          onClick={[Function]}
        >
          <WithStyles(TableCell)
            className="WidgetTable-textBody-10"
            key="1,0"
            padding="none"
          >
            <WithTheme(TableWidgetSelector)
              columnMeta={
                Object {
                  "label": "Alarm state",
                  "tags": Array [
                    "info:alarm",
                  ],
                }
              }
              columnWidgetTag="info:alarm"
              rowChangeHandler={[Function]}
              rowPath={
                Object {
                  "column": 0,
                  "label": "alarm",
                  "row": 1,
                }
              }
              setFlag={[Function]}
              value={1}
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            className="WidgetTable-textBody-10"
            key="1,1"
            padding="none"
          >
            <WithTheme(TableWidgetSelector)
              columnMeta={
                Object {
                  "label": "Time set",
                  "tags": Array [
                    "widget:textupdate",
                  ],
                }
              }
              columnWidgetTag="widget:textupdate"
              rowChangeHandler={[Function]}
              rowPath={
                Object {
                  "column": 1,
                  "label": "timeStamp",
                  "row": 1,
                }
              }
              setFlag={[Function]}
              value="1970-01-01T00:00:02.000Z"
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            className="WidgetTable-textBody-10"
            key="1,2"
            padding="none"
          >
            <WithTheme(TableWidgetSelector)
              columnMeta={
                Object {
                  "label": "Value",
                  "tags": Array [
                    "widget:textupdate",
                  ],
                }
              }
              columnWidgetTag="widget:textupdate"
              rowChangeHandler={[Function]}
              rowPath={
                Object {
                  "column": 2,
                  "label": "value",
                  "row": 1,
                }
              }
              setFlag={[Function]}
              value={-1}
            />
          </WithStyles(TableCell)>
        </WithStyles(TableRow)>
        <WithStyles(TableRow)
          className="WidgetTable-rowFormat-6"
          key="2"
          onClick={[Function]}
        >
          <WithStyles(TableCell)
            className="WidgetTable-textBody-10"
            key="2,0"
            padding="none"
          >
            <WithTheme(TableWidgetSelector)
              columnMeta={
                Object {
                  "label": "Alarm state",
                  "tags": Array [
                    "info:alarm",
                  ],
                }
              }
              columnWidgetTag="info:alarm"
              rowChangeHandler={[Function]}
              rowPath={
                Object {
                  "column": 0,
                  "label": "alarm",
                  "row": 2,
                }
              }
              setFlag={[Function]}
              value={0}
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            className="WidgetTable-textBody-10"
            key="2,1"
            padding="none"
          >
            <WithTheme(TableWidgetSelector)
              columnMeta={
                Object {
                  "label": "Time set",
                  "tags": Array [
                    "widget:textupdate",
                  ],
                }
              }
              columnWidgetTag="widget:textupdate"
              rowChangeHandler={[Function]}
              rowPath={
                Object {
                  "column": 1,
                  "label": "timeStamp",
                  "row": 2,
                }
              }
              setFlag={[Function]}
              value="1970-01-01T00:00:03.000Z"
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            className="WidgetTable-textBody-10"
            key="2,2"
            padding="none"
          >
            <WithTheme(TableWidgetSelector)
              columnMeta={
                Object {
                  "label": "Value",
                  "tags": Array [
                    "widget:textupdate",
                  ],
                }
              }
              columnWidgetTag="widget:textupdate"
              rowChangeHandler={[Function]}
              rowPath={
                Object {
                  "column": 2,
                  "label": "value",
                  "row": 2,
                }
              }
              setFlag={[Function]}
              value={false}
            />
          </WithStyles(TableCell)>
        </WithStyles(TableRow)>
      </WithStyles(TableBody)>
    </WithStyles(Table)>
    <WithStyles(Table)>
      <WithStyles(TableFooter) />
    </WithStyles(Table)>
  </div>
  <WithStyles(Table)
    className="WidgetTable-footerLayout-2"
  >
    <WithStyles(TableFooter)>
      <WithStyles(TableRow)
        className="WidgetTable-rowFormat-6"
      />
    </WithStyles(TableFooter)>
  </WithStyles(Table)>
</div>
`;
